art/steps.py

```diff
@@ -395,6 +402,8 @@ def do(self, previous_states: Dict):
         Args:
             previous_states (Dict): previous states
         """
+        # Running setup for ml_train with pure train dataloader
+        self.datamodule.setup(stage=TrainingStage.TRAIN.value)
```

Comment:
This doesn't scale well to new steps. My idea would be like this

add `get_trainloader()` Function to the base class Inside which you have try catch statement. If it suceeds it just return train_dataloader if it fails it runs datamodule setup function and then returns train_dataloader